sdks/java/core/src/main/java/org/apache/beam/sdk/io/PubsubUnboundedSource.java

```diff
@@ -292,6 +296,17 @@ public void finalizeCheckpoint() throws IOException {
     }
 
     /**
+     * Return current time according to {@code reader}.
+     */
+    private static long now(PubsubReader reader) {
+      if (reader.outer.outer.clock == null) {
+        return System.currentTimeMillis();
```

Comment:
It's not serializable, so I either had to roll my own or avoid passing Clock.SYSTEM.